### DISCUSSION TITLE
Ideas for finding class/instance for a window

### DISCUSSION THREAD

**Discussion by AckslD** [herbstluftwm/herbstluftwm#1442] [Q&A]
`zoom` sometimes opens a small window dialog when entering a meeting. I would like this to be floating but don't know what the class/instance to use for the rule. What's a good way to find this out?

I added the following keybind to my config in an attempt to quickly get this information:
```
hc keybind $Mod-Shift-t spawn notify-send 'hlwm' "$(herbstclient foreach C clients attr C | rg 'class|instance|title =')"
```
but still haven't managed to figure out which window is which when triggering this command.

I also tried
```
hc keybind $Mod-Shift-t spawn notify-send 'hlwm' "$(herbstclient attr clients.focus)"
```
but that just gave info about a `kitty` window which is not the focused one for some reason.

**Comment by tcoopman** [comment]
@AckslD I know this is a long time later, but did you figure out how to do this?

**Comment by fabian-thomas** [comment]
You can probably use `hc --idle` and print the classes of all newly spawned windows.

But if you are only interested in zoom, here is what I have:
```sh
hc rule class=zoom title~".*Chat.*" floating=on
hc rule class=zoom title~".*Participants.*" floating=on
hc rule class=zoom title~".*Settings.*" floating=on

# looks like all popups have lower-case zoom as title
hc rule title~zoom floating=on
```
